
The "drop print" commit removed the whole branch and not just the print.
For some ARM64 cpus, this leads to hard lockup when CONFIG_MITIGATE_SPECTRE_BRANCH_HISTORY is not enabled.

Fixes: 62e72463ca71 ("arm64: proton-pack: Drop print when !CONFIG_MITIGATE_SPECTRE_BRANCH_HISTORY")

diff --speed-large-files --no-dereference --minimal -Naur linux-6.18-rc6/arch/arm64/kernel/proton-pack.c linux-6.18-rc6/arch/arm64/kernel/proton-pack.c
--- linux-6.18-rc6/arch/arm64/kernel/proton-pack.c	2025-11-16 23:25:38.000000000 +0100
+++ linux-6.18-rc6/arch/arm64/kernel/proton-pack.c	2025-11-22 14:15:40.762119879 +0100
@@ -1032,6 +1032,8 @@
 
 	if (arm64_get_spectre_v2_state() == SPECTRE_VULNERABLE) {
 		/* No point mitigating Spectre-BHB alone. */
+	} else if (!IS_ENABLED(CONFIG_MITIGATE_SPECTRE_BRANCH_HISTORY)) {
+		/* Do nothing */
 	} else if (supports_ecbhb(SCOPE_LOCAL_CPU)) {
 		state = SPECTRE_MITIGATED;
 		set_bit(BHB_HW, &system_bhb_mitigations);
